dynamic-proxy/src/main/java/com/iluwatar/dynamicproxy/tinyrestclient/TinyRestClient.java

```diff
@@ -0,0 +1,157 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.dynamicproxy.tinyrestclient;
+
+import com.iluwatar.dynamicproxy.tinyrestclient.annotation.Body;
+import com.iluwatar.dynamicproxy.tinyrestclient.annotation.HttpMethod;
+import com.iluwatar.dynamicproxy.tinyrestclient.annotation.Path;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Arrays;
+
+/**
+ * Class to handle all the http communication with a Rest API.
+ * It is supported by the HttpClient Java library.
+ */
+public class TinyRestClient {
+
+  private String baseUrl;
+  private HttpClient httpClient;
+
+  /**
+   * Class constructor.
+   *
+   * @param baseUrl    Root url for endpoints.
+   * @param httpClient Handle the http communication.
+   */
+  public TinyRestClient(String baseUrl, HttpClient httpClient) {
+    this.baseUrl = baseUrl;
+    this.httpClient = httpClient;
+  }
+
+  /**
+   * Creates a http communication to request and receive data from an endpoint.
+   *
+   * @param method Interface's method which is annotated with a http method.
+   * @param args   Method's arguments passed in the call.
+   * @return Response from the endpoint.
+   * @throws IOException          Exception thrown when any fail happens in the call.
+   * @throws InterruptedException Exception thrown when call is interrupted.
+   */
+  public Object send(Method method, Object[] args) throws IOException, InterruptedException {
+    var httpMethodAnnotation = getHttpMethodAnnotation(method.getDeclaredAnnotations());
+    if (httpMethodAnnotation == null) {
+      return  null;
+    }
+    var httpMethodName = httpMethodAnnotation.annotationType().getSimpleName().toUpperCase();
+    var url = baseUrl + buildUrl(method, args, httpMethodAnnotation);
+    var bodyPublisher = buildBodyPublisher(method, args);
+    var httpRequest = HttpRequest.newBuilder()
+        .uri(URI.create(url))
+        .header("Content-Type", "application/json")
+        .method(httpMethodName, bodyPublisher)
+        .build();
+    var httpResponse = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+    return getResponse(method, httpResponse);
+  }
+
+  private String buildUrl(Method method, Object[] args, Annotation httpMethodAnnotation) {
+    var url = annotationValue(httpMethodAnnotation);
+    if (url == null) {
+      return "";
+    }
+    var index = 0;
+    for (var parameter : method.getParameters()) {
+      var pathAnnotation = getAnnotationOf(parameter.getDeclaredAnnotations(), Path.class);
+      if (pathAnnotation != null) {
+        var pathParam = "{" + annotationValue(pathAnnotation) + "}";
```

Comment:
maybe we can consider encoding path parameters to handle special characters.

Reply:
Done. Encoding for pathValue was added.